onnx/defs/nn/defs.cc

```diff
@@ -1416,4 +1402,68 @@ ONNX_OPERATOR_SET_SCHEMA(
         .SetDoc(LRN_ver1_doc)
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
+static const char* StringNormalizer_ver9_doc = R"DOC(
+[optional] Step1: Remove elements in X if they match any of the stop words so
+ that the output tensor will not contain any stop words. This operator only accepts [C]-
+ and [1, C]-tensors. If all elements in X are dropped, the output will be the default value
+ of string tensor with shape [1] if input shape is [C] and shape [1, 1] if input shape is [1, C].
+[optional] Step2: Lower all characters (if action is LOWER) in X or capitalize them (when action is UPPER).
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    StringNormalizer,
+    9,
+    OpSchema()
+        .Input(0, "X", "Strings to normalize", "T")
+        .Output(0, "Y", "Normalized strings", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(string)"},
+            "Input/Output is a UTF-8 string tensor")
+        .Attr(
+            "casechangeaction",
+            "string enum that cases output to be lowercased/uppercases/unchanged. Valid values are \"LOWER\", \"UPPER\", \"NONE\"",
+            AttributeProto::STRING)
+        .Attr(
+            "is_case_sensitive",
+            "Boolean. Whether the identification of stop words in X is case-sensitive.",
+            AttributeProto::INT)
+        .Attr(
+            "stopwords",
+            "List of stop words",
+            AttributeProto::STRINGS,
+            OPTIONAL)
+        .Attr(
+            "locale",
+            "Environment dependent string that denotes the locale according to which output strings needs to be upper/lowercased."
+            "Default en_US or platform specific equivalent",
+            AttributeProto::STRING,
+            OPTIONAL)
+        .SetDoc(StringNormalizer_ver9_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          auto output_elem_type = ctx.getOutputType(0)->mutable_tensor_type();
+          output_elem_type->set_elem_type(TensorProto::STRING);
+          if (hasInputShape(ctx, 0)) {
+            TensorShapeProto output_shape;
+            auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+            auto dim_size = input_shape.dim_size();
+            // Last axis dimension is unknown if we have stop-words since we do
+            // not know how many stop-words are dropped
+            if (dim_size == 0) {
+              output_shape.add_dim();
+            } else if (dim_size == 1) {
+              // Copy C if no stopwords otherwise unknown
+              output_shape.add_dim();
+            } else if (dim_size == 2) {
+              // Copy B-dim
+              *output_shape.add_dim() = input_shape.dim(0);
```

Comment:
In this case, the first dim value has to be "1", right? per the description above. if yes, check it please.

Reply:
In case of 2-dim the first value can be an int, dim_param or nothing, so I simply copy it.